pkg/ring/kv/etcd/etcd.go

```diff
@@ -16,9 +18,13 @@ import (
 
 // Config for a new etcd.Client.
 type Config struct {
-	Endpoints   []string      `yaml:"endpoints"`
-	DialTimeout time.Duration `yaml:"dial_timeout"`
-	MaxRetries  int           `yaml:"max_retries"`
+	Endpoints          []string      `yaml:"endpoints"`
+	DialTimeout        time.Duration `yaml:"dial_timeout"`
+	MaxRetries         int           `yaml:"max_retries"`
+	CertFile           string        `yaml:"cert_file"`
```

Comment:
I would add the `tls_` prefix to all such config options and `tls-` to CLI flags.